src/services/symbolication.rs

```diff
@@ -1642,70 +1693,55 @@ impl SymbolicationActor {
-                    let mut stacktraces = Vec::with_capacity(threads.len());
-                    for (index, thread) in threads.iter().enumerate() {
-                        let registers = match thread.frames().get(0) {
-                            Some(frame) => map_symbolic_registers(
-                                frame.registers(minidump_state.system_info.cpu_arch),
-                            ),
-                            None => Registers::new(),
-                        };
-
-                        // Trim infinite recursions explicitly because those do not
-                        // correlate to minidump size. Every other kind of bloated
-                        // input data we know is already trimmed/rejected by raw
-                        // byte size alone.
-                        let frame_count = thread.frames().len().min(20000);
-                        let mut frames = Vec::with_capacity(frame_count);
-                        for frame in thread.frames().iter().take(frame_count) {
-                            let return_address =
-                                frame.return_address(minidump_state.system_info.cpu_arch);
-                            module_builder.mark_referenced(return_address);
-
-                            frames.push(RawFrame {
-                                instruction_addr: HexValue(return_address),
-                                package: frame.module().map(CodeModule::code_file),
-                                trust: frame.trust(),
-                                ..RawFrame::default()
-                            });
+                    let timer_old = Instant::now();
+                    let process_state =
+                        ProcessState::from_minidump_breakpad(&minidump, Some(&cfi))?;
+                    let timer_old = timer_old.elapsed();
+                    let (object_infos, stacktraces, minidump_state) = Self::post_process(process_state, cfi_caches_cloned);
+
+                    // Run and time the new stackwalking method if compare_stackwalking_methods was passed
+                    if compare_stackwalking_methods {
+                        let timer_new = Instant::now();
+                        match ProcessState::from_minidump(&minidump, Some(&cfi)) {
+                            Ok(process_state_new) => {
+                                let timer_new = timer_new.elapsed();
+                                let (object_infos_new, stacktraces_new, _) = Self::post_process(process_state_new, cfi_caches);
+
+                                // Report whether the stacktraces are different or identical.
+                                if stacktraces == stacktraces_new && object_infos == object_infos_new {
+                                    metric!(counter("minidump.stackwalk.identical") += 1);
+                                } else {
+                                    metric!(counter("minidump.stackwalk.different") += 1);
```

Comment:
don't we also want to be able to debug this now?  How do we get enough information to do this comparison?  dump it all in a sentry event?  I think you could do `sentry::with_scope(|scope| scope.set_extra(...))` to add info to the event before capturing, but I'm fairly sure this is abuse of sentry so we may need to do something else.
/cc @jan-auer any ideas on better ways to log this?

Reply:
I think we can get just numbers first, and depending on how they look, we can start capturing the scope and also error info below.